.infrastructure/helm-chart/todoapp/templates/deployment.yml

```diff
@@ -0,0 +1,86 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.Chart.Name}}
+  namespace: {{.Values.todoapp.namespace}}
+spec:
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: {{.Values.todoapp.rollingUpdate.maxSurge}}
+      maxUnavailable: {{.Values.todoapp.rollingUpdate.maxUnavailable}}
+  selector:
+    matchLabels:
+      app: {{.Chart.Name}}
+  template:
+    metadata:
+      labels:
+        app: {{.Chart.Name}}
+    spec:
+      serviceAccountName: {{.Values.todoapp.RBAC.name}}
```

Comment:
Ensure that the `todoapp.RBAC.name` value is defined in your `values.yaml` file. If it's not defined, this could lead to issues when deploying the chart. Consider providing a default value or handling the case where it might be undefined.